packages/plugin-axe/src/lib/meta/groups.unit.test.ts

```diff
@@ -0,0 +1,128 @@
+import type { RuleMetadata } from 'axe-core';
+import { describe, expect, it } from 'vitest';
+import { loadAxeRules, transformRulesToGroups } from './transform.js';
+
+describe('transformRulesToGroups', () => {
+  describe('wcag21aa preset', () => {
+    it('should create WCAG 2.1 Level A and AA groups', () => {
+      const groups = transformRulesToGroups(
+        loadAxeRules('wcag21aa'),
+        'wcag21aa',
+      );
+
+      expect(groups.map(({ slug }) => slug)).toEqual([
+        'wcag21-level-a',
+        'wcag21-level-aa',
+      ]);
+      expect(groups.map(({ title }) => title)).toEqual([
+        'WCAG 2.1 Level A',
+        'WCAG 2.1 Level AA',
+      ]);
+    });
+
+    it('should have refs in WCAG groups', () => {
+      transformRulesToGroups(loadAxeRules('wcag21aa'), 'wcag21aa').forEach(
+        ({ refs }) => {
+          expect(refs.length).toBeGreaterThan(0);
+        },
+      );
```

Comment:
Loop-based assertions are not a very good practice. The reason is that it's not transparent how many assertions are being run, and it's easy to miss when a test isn't actually asserting anything (empty array).

It's possible to handle such cases with something like [`expect.assertions(<count>)`](https://vitest.dev/api/expect.html#expect-assertions). But in this case, I'd consider that overkill. You know there are precisely 2 items in the returned array, so you check each index explicitly:

```suggestion
      const groups = transformRulesToGroups(
        loadAxeRules('wcag21aa'),
        'wcag21aa',
      );

      expect(groups[0]!.refs.length).toBeGreaterThan(0);
      expect(groups[1]!.refs.length).toBeGreaterThan(0);
```